presto-main/src/main/java/com/facebook/presto/operator/NativeExecutionOperator.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.common.Page;
+import com.facebook.presto.execution.ScheduledSplit;
+import com.facebook.presto.execution.TaskSource;
+import com.facebook.presto.execution.buffer.PagesSerdeFactory;
+import com.facebook.presto.execution.scheduler.TableWriteInfo;
+import com.facebook.presto.memory.context.LocalMemoryContext;
+import com.facebook.presto.spi.UpdatablePageSource;
+import com.facebook.presto.spi.page.PagesSerde;
+import com.facebook.presto.spi.plan.PlanNodeId;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+/*
+ * TODO: The lifecycle of NativeExecutionOperator is:
+ *  1. Launch the native engine external process when initialing the operator.
+ *  2. Serialize and pass the planFragment, tableWriteInfo and taskSource to the external process through NativeExecutionTask APIs.
+ *  3. Block on NativeExecutionTask's pollResult call to retrieve the data in {@link SerializedPage} page format back external process.
+ *  4. Deserialize the serialized page to presto Page and return it back to driver from the getOutput method.
+ *  5. The close() will be called by the driver when the pollResult returns empty and the Future of NativeExecutionTask has completed.
+ *  6. Shut down the external process upon calling of close() method
+ */
+public class NativeExecutionOperator
+        implements SourceOperator
+{
+    private final PlanNodeId sourceId;
+    private final OperatorContext operatorContext;
+    private final LocalMemoryContext systemMemoryContext;
+    private final PlanFragment planFragment;
+    private final TableWriteInfo tableWriteInfo;
```

Comment:
Do we need `tableWriteInfo`? Would be good to add a todo to show these 3 (or 2) fields will be used in some future commit

Reply:
`TableWriteInfo` will be used to create the `[NativeExecutionTask](https://github.com/prestodb/presto/blob/6b0202c0c3a45a17c734261e8c9b51c57b82f8d3/presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTaskFactory.java#L67)`, added the Todo as suggested.